Fixed StackOverflow in WebappClassLoader.getURLs + removed workaround

dmatej opened the pull request:
Fixed SO
- seen in tests when logging used toString after the CL closed.
- Now when it is used after the closure, returns an empty array instead of exception

Removed clearReferencesRmiTargets
- Not required any more; fixes test failure mentioned in comments
- The memory leak was fixed in JDK9
- See https://bugs.openjdk.org/browse/JDK-8157570

dmatej committed:
Fixed StackOverflow in WebappClassLoader.getURLs

- seen in tests when logging used toString after the CL closed.
- Now when it is used after the closure, returns an empty array
  instead of exception

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

arjantijms commented:
Failed with:

```
10:39:41  [ERROR]   WebappClassLoaderTest.getURLs:116 » InaccessibleObject Unable to make field private final java.lang.ClassLoader sun.rmi.transport.Target.ccl accessible: module java.rmi does not "opens sun.rmi.transport" to unnamed module @314c508a
10:39:41  [INFO] 
10:39:41  [ERROR] Tests run: 4, Failures: 0, Errors: 1, Skipped: 0
```

dmatej commented:
> Failed with:
> 
> ```
> 10:39:41  [ERROR]   WebappClassLoaderTest.getURLs:116 » InaccessibleObject Unable to make field private final java.lang.ClassLoader sun.rmi.transport.Target.ccl accessible: module java.rmi does not "opens sun.rmi.transport" to unnamed module @314c508a
> 10:39:41  [INFO] 
> 10:39:41  [ERROR] Tests run: 4, Failures: 0, Errors: 1, Skipped: 0
> ```

Where is it from? This is not the failure from jenkins, there failed logging test because of the timing issue.

dmatej committed:
Fixed exception when using reflection to clean webappcl reference

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Removed clearReferencesRmiTargets

- This leak was already fixed in JDK9
- see https://bugs.openjdk.org/browse/JDK-8157570

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>
